### PR TITLE
Refactor iter_sent.hpp to make structs lowercase

### LINKED ISSUE #5075: Make the structs in all iter_sent.hpp lower case

### PR DESCRIPTION
**Pull request by sudo-panda**
Fixes #5075 

### Changes
- Change Iterator struct to iterator
- Change Sentinel struct to sentinel
- Cleanup in files that had unnecessary includes

### CODE REVIEW

**Review by hkaiser — DISMISSED** [review]
LGTM, thanks!

**Comment by msimberg** [comment]
retest

**Comment by sudo-panda** [comment]
> retest

@msimberg is there a way to trigger it manually without pushing to the branch again?

File: libs/parallelism/algorithms/tests/regressions/count_3646.cpp [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 #include "iter_sent.hpp"
 
-struct BitCountingIterator : public Iterator<std::int64_t>
+struct BitCountingIterator : public iterator<std::int64_t>
```
Comment (msimberg) [inline libs/parallelism/algorithms/tests/regressions/count_3646.cpp:19]: I realized this wasn't part of the original issue, but would you mind changing `BitCountingIterator` to `bit_counting_iterator`? In general there are a few other names in these files that are in camel case, that should be changed to snake case.

File: libs/parallelism/algorithms/tests/regressions/reduce_3641.cpp [hunk 2]
```diff
@@ -8,24 +8,21 @@
 // #3641: Trouble with using ranges-v3 and hpx::parallel::reduce
 // #3646: Parallel algorithms should accept iterator/sentinel pairs
 
-#include <hpx/hpx_main.hpp>
```
Comment (msimberg) [inline libs/parallelism/algorithms/tests/regressions/reduce_3641.cpp:11]: Please don't remove this header. Including it ensures that `main` is started on the HPX runtime.

File: libs/parallelism/algorithms/tests/unit/container_algorithms/foreach_adapt.cpp [hunk 3]
```diff
@@ -4,16 +4,12 @@
 //  Distributed under the Boost Software License, Version 1.0. (See accompanying
 //  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
-#include <hpx/hpx_main.hpp>
```
Comment (msimberg) [inline libs/parallelism/algorithms/tests/unit/container_algorithms/foreach_adapt.cpp:7]: Same here, please keep this header.

**Review by msimberg — CHANGES_REQUESTED** [review]
Thanks! A couple of minor comments.

**Comment by msimberg** [comment]
> @msimberg is there a way to trigger it manually without pushing to the branch again?

That was it (i.e. a comment with `retest`), but that is only for whitelisted members (to avoid allowing anyone opening a PR to be able to run on our jenkins CI).

**Comment by aurianer** [comment]
Thanks for working on this @sudo-panda!

File: libs/parallelism/algorithms/tests/regressions/count_3646.cpp (hunk 1 above)
Comment (sudo-panda) [reply libs/parallelism/algorithms/tests/regressions/count_3646.cpp:19]: I would be happy to do so. Can you please tell me the directory in which these changes are required or is it in the whole repo?

Comment (msimberg) [reply libs/parallelism/algorithms/tests/regressions/count_3646.cpp:19]: It's enough if you do so on the files that you've touched so far in this PR.

Comment (sudo-panda) [reply libs/parallelism/algorithms/tests/regressions/count_3646.cpp:19]: Sure!

Comment (sudo-panda) [reply libs/parallelism/algorithms/tests/regressions/count_3646.cpp]: There was just that one(`BitCountingIterator`) in the files I have touched.

File: libs/parallelism/algorithms/tests/regressions/reduce_3641.cpp [hunk 4]
```diff
@@ -13,18 +13,16 @@
...
 
-#include <cstddef>
 #include <cstdint>
-#include <iterator>
 
 int main()
 {
     std::int64_t result = hpx::ranges::reduce(hpx::execution::seq,
-        Iterator<std::int64_t>{0}, Sentinel<int64_t>{100}, std::int64_t(0));
+        iterator<std::int64_t>{0}, Sentinel<int64_t>{100}, std::int64_t(0));
```
Comment (msimberg) [inline libs/parallelism/algorithms/tests/regressions/reduce_3641.cpp:21]: Please check that the files you've changed build for you:
```suggestion
        iterator<std::int64_t>{0}, sentinel<int64_t>{100}, std::int64_t(0));
```

File: libs/parallelism/algorithms/tests/unit/container_algorithms/foreach_adapt.cpp [hunk 5]
```diff
@@ -22,25 +19,25 @@ void myfunction(int i)
 
 void test_invoke_projected()
 {
-    Iterator<std::int64_t> iter = hpx::ranges::for_each(hpx::execution::seq,
-        Iterator<std::int64_t>{0}, Sentinel<int64_t>{100}, myfunction);
+    iterator<std::int64_t> iter = hpx::ranges::for_each(hpx::execution::seq,
+        iterator<std::int64_t>{0}, Sentinel<int64_t>{100}, myfunction);
```
Comment (msimberg) [inline libs/parallelism/algorithms/tests/unit/container_algorithms/foreach_adapt.cpp:23]: Same here:
```suggestion
        iterator<std::int64_t>{0}, sentinel<int64_t>{100}, myfunction);
```

File: libs/parallelism/algorithms/tests/regressions/reduce_3641.cpp (hunk 4 above)
Comment (sudo-panda) [reply libs/parallelism/algorithms/tests/regressions/reduce_3641.cpp]: Sorry about that.

**Comment by msimberg** [comment]
retest

**Pull request merged**